### ISSUE TITLE
Clarify which software license applies to which source files: GPLv3 vs LGPLv3

### ISSUE THREAD

**Issue by dozylynx** [libvmi/libvmi#749]
Both GPLv3 and LGPLv3 software licenses are present in the LibVMI source tree, but it not clear which license applies to which LibVMI source code files.

This presents a problem for downstream packaging of the library and it is unclear whether any binary linking against LibVMI must be licensed under the terms of the GPLv3.

Hopefully there is consensus on the correct licensing among LibVMI contributors, so clarification will just involve a simple addition to the docs.

**Comment by tklengyel** [comment]
Could you point out where you see GPLv3 in the tree?

**Comment by smaresca** [comment]
The library itself is LGPLv3 in its entirety, as expressed consistently in headers.  This has been the case since the initial public release of LibVMI by Bryan Payne.  Please let us know if there is any specific ambiguity.

The only file explicitly released under GPLv3 is config_editor.py, a contributed script not included in the compiled objects nor used in their creation. This does not impact the library license.  

COPYING and COPYING.LESSER are both included as a result.

**Comment by dozylynx** [comment]
Thanks for the responses. As you've indicated, the software license (LGPLv3) is consistently indicated in the C source code files -- I should've looked closer there before posting, sorry. My confusion arose from a couple of things:
* The presence of both licenses in the top level directory, with no statement in either of the READMEs indicating which license applies where.
* The COPYING file containing GPLv3: I'd expected that file to contain the license that applied to the majority of the source, which it isn't in this case here.

Are the Makefiles also licensed as LGPLv3?

A separate minor thing I noticed: `tools/style/checkpatch.pl` is indicated as distributed under the terms of GPLv2, but there's no copy of that license present.

**Comment by tklengyel** [comment]
> tools/style/checkpatch.pl

I don't think this script it actually used for anything anymore so I'll just remove it
